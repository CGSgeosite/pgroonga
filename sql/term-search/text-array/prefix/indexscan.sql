CREATE TABLE tags (
  names text[]
);

INSERT INTO tags VALUES (ARRAY['PostgreSQL', 'PG']);
INSERT INTO tags VALUES (ARRAY['Groonga', 'grn']);
INSERT INTO tags VALUES (ARRAY['PGroonga', 'pgrn']);

CREATE INDEX pgrn_index ON tags
  USING pgroonga (names pgroonga.text_array_term_search_ops_v2);

SET enable_seqscan = off;
SET enable_indexscan = on;
SET enable_bitmapscan = off;

SELECT names
  FROM tags
 WHERE names &^> 'pG';

DROP TABLE tags;
