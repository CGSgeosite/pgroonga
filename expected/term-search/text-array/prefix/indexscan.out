CREATE TABLE tags (
  names text[]
);
INSERT INTO tags VALUES (ARRAY['PostgreSQL', 'PG']);
INSERT INTO tags VALUES (ARRAY['Groonga', 'grn']);
INSERT INTO tags VALUES (ARRAY['PGroonga', 'pgrn']);
INSERT INTO tags VALUES (ARRAY[]::text[]);
CREATE INDEX pgrn_index ON tags
  USING pgroonga (names pgroonga_text_array_term_search_ops_v2);
SET enable_seqscan = off;
SET enable_indexscan = on;
SET enable_bitmapscan = off;
EXPLAIN (COSTS OFF)
SELECT names
  FROM tags
 WHERE names &^ 'pG';
             QUERY PLAN              
-------------------------------------
 Index Scan using pgrn_index on tags
   Index Cond: (names &^ 'pG'::text)
(2 rows)

SELECT names
  FROM tags
 WHERE names &^ 'pG';
      names      
-----------------
 {PGroonga,pgrn}
 {PostgreSQL,PG}
(2 rows)

DROP TABLE tags;
